data_structures/rtree.hpp

```diff
@@ -29,7 +29,8 @@ namespace spatial {
                     void pick_seeds(std::vector<Entry> const& entry_choices);
                     void distribute(std::vector<Entry>& leftovers);
                     int pick_next(std::vector<Entry> const& leftovers) const;
-                    bool is_leaf() const;
+                    bool is_leaf() const; 
+                    bool check_load() const;
```

Comment:
More of a tangential design consideration to think about, but [it is often argued](https://anthonysciamanna.com/2016/02/14/should-private-methods-be-tested.html) that you should always be able to test correctness fully with your existing interface. When you need to change the interface just for testing (e.g., make a private method public or a add a new method altogether in order to expose state), it suggests that your class may have too many responsibilities.